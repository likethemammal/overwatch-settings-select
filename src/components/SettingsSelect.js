import React, { Component } from 'react'
import _ from 'underscore'

import Menu  from 'material-ui/Menu'

import DownSVG from 'svg-icon/dist/svg/mfglabs/chevron_down.svg'

import general from '@likethemammal/general-components'


const { SVG } = general.components

import css from './SettingsSelect.less'

import {
    selectedOption as _selectedOption,
} from '../selectors/select'

const menuOverrideProps = {
    marginThreshold: 0,
    getContentAnchorEl: null,
    anchorOrigin: {
        vertical: 'bottom',
        horizontal: 'left',
    },
    MenuListProps: {
        disablePadding: true,
        className: css.list_override,
        classes: {
            root: css.list_root_override,
        }
    },
    PaperProps: {
        style: {
            background: 'transparent'
        }
    }
}

export default class SettingsSelect extends Component {

    state = {
        open: false,
        anchorEl: null,
    }

    onMenuOpen = (ev) => {
        this.setState({ open: true, anchorEl: ev.currentTarget })

        ev.target.blur()
    }

    onMenuClose = (ev) => {
        this.setState({ open: false, })
    }

    onSelect = (index) => {
        this.setState({ open: false })

        this.props.onChange(index)
    }

    render() {
        const { options, label, selectedIndex } = this.props

        const {
            open,
            anchorEl,
        } = this.state

        const selectedOption = _selectedOption(this.props)

        return <div className={css.container}>
            <div className={css.inner}>

                <div
                    className={!open ? css.label : css.label__selected}
                >
                    {label}
                </div>

                <div className={css.select}>

                    <button
                        onClick={this.onMenuOpen}
                        className={css.active}
                    >
                        {selectedOption}
                        <div className={css.icon_container}>
                            <SVG
                                raw={DownSVG}
                                className={css.icon}
                            />
                        </div>
                    </button>

                    <Menu
                        {...menuOverrideProps}
                        id={label}
                        open={open}
                        anchorEl={anchorEl}
                        onBackdropClick={this.onMenuClose}
                    >
                        {_.map(options, (option, i) => {
                            return <li
                                onClick={this.onSelect.bind(null, i)}
                                key={i}
                                className={css.list_item}
                                selected={i === selectedIndex}
                            >
                                {option}
                            </li>
                        })}
                    </Menu>
                </div>
            </div>
        </div>
    }
}
