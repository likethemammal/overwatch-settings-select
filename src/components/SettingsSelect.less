@import "../../node_modules/@likethemammal/overwatch-general/lib/colors.less";

@item_width: 377px;

@row_hover_background: hsla(0, 0%, 15%, 0.25);
@row_height: 3.8rem;

.container {
    padding: 3px 0 4px;

    &:hover {
        .label {
            background: @row_hover_background;
        }
    }

    //low browser support, so cant be combined with :hover selector
    &:focus-within {
        .label {
            background: @row_hover_background;
        }
    }
}

.inner {
    position: relative;
}

.__text {
    composes: _font_general from '@likethemammal/overwatch-general/lib/general.css';
    line-height: 3.9rem;
    color: hsl(0, 0%, 90%);
}

.label {
    composes: __text;
    composes: cf from '@likethemammal/general-components/bin/general.css';
    padding-left: 22px;
    background: @black_faded_setting;
    transition: background 100ms ease-in;
    font-size: 2.3rem;
    height: @row_height;
}

.label__selected {
    composes: label;
    background: @row_hover_background;
}

.select {
    position: absolute;
    top: 0;
    right: 0;
}

.list_override {
    padding-right: 0 !important;
    margin-top: 2px !important;
}

.list_root_override {
    width: 100% !important;
}

.item {
    composes: __text;
    width: @item_width;
    line-height: 3.7rem;
    padding-left: 15px;
    letter-spacing: 0.01rem;
    font-size: 2.2rem;
    cursor: pointer;
    box-sizing: border-box;
}

.list_item {
    composes: item;
    background: hsl(220, 42%, 17%);
    padding: 10px 10px 0 11px;

    &:last-child {
        padding-bottom: 11px;
    }

    &:focus {
        border: 0;
        outline: 0;
    }

    &:hover, &:focus {
        .list_item_inner {
            background: hsl(196, 69%, 41%);
        }
    }
}

.list_item_inner {
    padding-left: 10px;
    border-radius: 3px;
    box-sizing: border-box;
}

.active {
    composes: item;
    background: @blue_faded_hover;
    border: 0;
    text-align: left;
    height: @row_height;

    &:focus {
        border: 0;
        outline: 0;
    }
}

.icon_container {
    composes: _icon_container from '@likethemammal/overwatch-general/lib/general.css';
    position: absolute;
    right: 5px;
    left: auto;
    top: 0px;

    height: 25px;
    width: 25px;
    padding: 5px 7px;
}
