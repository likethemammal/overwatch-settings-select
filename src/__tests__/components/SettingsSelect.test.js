import React from 'react';
import {shallow} from 'enzyme';

import general from '@likethemammal/general-components'

const shared = general.__tests__._shared.tests

import SettingsSelect from '../../../src'

const mock_options = ['Option 1', 'Option 2', 'Option 3', 'Option 4']

const mock_onChange = jest.fn()
const onChange = (index) => {
    changable_selectedIndex = index

    mock_onChange()
}

let changable_selectedIndex = 2


//--if theres only one option, error out

describe('SettingsSelect', () => {

    const component = shallow(
        <SettingsSelect
            onChange={onChange}
            selectedIndex={changable_selectedIndex}
            options={mock_options}
        />
    )

    shared.SNAPSHOTS_SHOULD_MATCH([
        component
    ])

    test('if onClick of menu button, open state should be true, Menu should receive open props', () => {
        expect(
            component.state('open')
        ).toEqual(false)

        component.find('button').simulate('click', {
            currentTarget: {},
            target: {
                blur: jest.fn()
            },
        })

        expect(
            component.state('open')
        ).toEqual(true)
    })

    test('if onClick of a <li/> item of Menu, open state should be false, onChange should be called', () => {
        expect(
            component.state('open')
        ).toEqual(true)

        component.find('li').at(0).simulate('click')

        expect(
            component.state('open')
        ).toEqual(false)
    })

    test('if selectedIndex is not zero, when li is clicked, onChange should be called and should return the index of the new selectedOption', () => {

        expect(
            changable_selectedIndex
        ).toEqual(0)

        component.find('li').at(1).simulate('click')

        expect(
            mock_onChange.mock.calls.length
        ).toEqual(2)

        expect(
            changable_selectedIndex
        ).toEqual(1)
    })
})