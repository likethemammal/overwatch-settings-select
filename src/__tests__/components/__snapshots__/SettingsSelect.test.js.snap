// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSelect Snapshot 0 should match 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SettingsSelect
    onChange={[Function]}
    options={
      Array [
        "Option 1",
        "Option 2",
        "Option 3",
        "Option 4",
      ]
    }
    selectedIndex={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        <div
          className={undefined}
        />
        <div
          className={undefined}
        >
          <button
            className={undefined}
            onClick={[Function]}
          >
            Option 3
            <div
              className={undefined}
            >
              <t
                className="SVG-container"
                raw="test-file-stub"
              />
            </div>
          </button>
          <WithStyles(Menu)
            MenuListProps={
              Object {
                "className": undefined,
                "classes": Object {
                  "root": undefined,
                },
                "disablePadding": true,
              }
            }
            PaperProps={
              Object {
                "style": Object {
                  "background": "transparent",
                },
              }
            }
            anchorEl={null}
            anchorOrigin={
              Object {
                "horizontal": "left",
                "vertical": "bottom",
              }
            }
            getContentAnchorEl={null}
            id={undefined}
            marginThreshold={0}
            onBackdropClick={[Function]}
            open={false}
          >
            <li
              className={undefined}
              onClick={[Function]}
              selected={false}
            >
              Option 1
            </li>
            <li
              className={undefined}
              onClick={[Function]}
              selected={false}
            >
              Option 2
            </li>
            <li
              className={undefined}
              onClick={[Function]}
              selected={true}
            >
              Option 3
            </li>
            <li
              className={undefined}
              onClick={[Function]}
              selected={false}
            >
              Option 4
            </li>
          </WithStyles(Menu)>
        </div>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          />,
          <div
            className={undefined}
          >
            <button
              className={undefined}
              onClick={[Function]}
            >
              Option 3
              <div
                className={undefined}
              >
                <t
                  className="SVG-container"
                  raw="test-file-stub"
                />
              </div>
            </button>
            <WithStyles(Menu)
              MenuListProps={
                Object {
                  "className": undefined,
                  "classes": Object {
                    "root": undefined,
                  },
                  "disablePadding": true,
                }
              }
              PaperProps={
                Object {
                  "style": Object {
                    "background": "transparent",
                  },
                }
              }
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "bottom",
                }
              }
              getContentAnchorEl={null}
              id={undefined}
              marginThreshold={0}
              onBackdropClick={[Function]}
              open={false}
            >
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 1
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 2
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={true}
              >
                Option 3
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 4
              </li>
            </WithStyles(Menu)>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className={undefined}
                onClick={[Function]}
              >
                Option 3
                <div
                  className={undefined}
                >
                  <t
                    className="SVG-container"
                    raw="test-file-stub"
                  />
                </div>
              </button>,
              <WithStyles(Menu)
                MenuListProps={
                  Object {
                    "className": undefined,
                    "classes": Object {
                      "root": undefined,
                    },
                    "disablePadding": true,
                  }
                }
                PaperProps={
                  Object {
                    "style": Object {
                      "background": "transparent",
                    },
                  }
                }
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  }
                }
                getContentAnchorEl={null}
                id={undefined}
                marginThreshold={0}
                onBackdropClick={[Function]}
                open={false}
              >
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 1
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 2
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={true}
                >
                  Option 3
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 4
                </li>
              </WithStyles(Menu)>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Option 3",
                  <div
                    className={undefined}
                  >
                    <t
                      className="SVG-container"
                      raw="test-file-stub"
                    />
                  </div>,
                ],
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Option 3",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <t
                      className="SVG-container"
                      raw="test-file-stub"
                    />,
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "SVG-container",
                      "raw": "test-file-stub",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "MenuListProps": Object {
                  "className": undefined,
                  "classes": Object {
                    "root": undefined,
                  },
                  "disablePadding": true,
                },
                "PaperProps": Object {
                  "style": Object {
                    "background": "transparent",
                  },
                },
                "anchorEl": null,
                "anchorOrigin": Object {
                  "horizontal": "left",
                  "vertical": "bottom",
                },
                "children": Array [
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 1
                  </li>,
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 2
                  </li>,
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={true}
                  >
                    Option 3
                  </li>,
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 4
                  </li>,
                ],
                "getContentAnchorEl": null,
                "id": undefined,
                "marginThreshold": 0,
                "onBackdropClick": [Function],
                "open": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Option 1",
                    "className": undefined,
                    "onClick": [Function],
                    "selected": false,
                  },
                  "ref": null,
                  "rendered": "Option 1",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Option 2",
                    "className": undefined,
                    "onClick": [Function],
                    "selected": false,
                  },
                  "ref": null,
                  "rendered": "Option 2",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Option 3",
                    "className": undefined,
                    "onClick": [Function],
                    "selected": true,
                  },
                  "ref": null,
                  "rendered": "Option 3",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Option 4",
                    "className": undefined,
                    "onClick": [Function],
                    "selected": false,
                  },
                  "ref": null,
                  "rendered": "Option 4",
                  "type": "li",
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          <div
            className={undefined}
          />
          <div
            className={undefined}
          >
            <button
              className={undefined}
              onClick={[Function]}
            >
              Option 3
              <div
                className={undefined}
              >
                <t
                  className="SVG-container"
                  raw="test-file-stub"
                />
              </div>
            </button>
            <WithStyles(Menu)
              MenuListProps={
                Object {
                  "className": undefined,
                  "classes": Object {
                    "root": undefined,
                  },
                  "disablePadding": true,
                }
              }
              PaperProps={
                Object {
                  "style": Object {
                    "background": "transparent",
                  },
                }
              }
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "bottom",
                }
              }
              getContentAnchorEl={null}
              id={undefined}
              marginThreshold={0}
              onBackdropClick={[Function]}
              open={false}
            >
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 1
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 2
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={true}
              >
                Option 3
              </li>
              <li
                className={undefined}
                onClick={[Function]}
                selected={false}
              >
                Option 4
              </li>
            </WithStyles(Menu)>
          </div>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
            />,
            <div
              className={undefined}
            >
              <button
                className={undefined}
                onClick={[Function]}
              >
                Option 3
                <div
                  className={undefined}
                >
                  <t
                    className="SVG-container"
                    raw="test-file-stub"
                  />
                </div>
              </button>
              <WithStyles(Menu)
                MenuListProps={
                  Object {
                    "className": undefined,
                    "classes": Object {
                      "root": undefined,
                    },
                    "disablePadding": true,
                  }
                }
                PaperProps={
                  Object {
                    "style": Object {
                      "background": "transparent",
                    },
                  }
                }
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  }
                }
                getContentAnchorEl={null}
                id={undefined}
                marginThreshold={0}
                onBackdropClick={[Function]}
                open={false}
              >
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 1
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 2
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={true}
                >
                  Option 3
                </li>
                <li
                  className={undefined}
                  onClick={[Function]}
                  selected={false}
                >
                  Option 4
                </li>
              </WithStyles(Menu)>
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className={undefined}
                  onClick={[Function]}
                >
                  Option 3
                  <div
                    className={undefined}
                  >
                    <t
                      className="SVG-container"
                      raw="test-file-stub"
                    />
                  </div>
                </button>,
                <WithStyles(Menu)
                  MenuListProps={
                    Object {
                      "className": undefined,
                      "classes": Object {
                        "root": undefined,
                      },
                      "disablePadding": true,
                    }
                  }
                  PaperProps={
                    Object {
                      "style": Object {
                        "background": "transparent",
                      },
                    }
                  }
                  anchorEl={null}
                  anchorOrigin={
                    Object {
                      "horizontal": "left",
                      "vertical": "bottom",
                    }
                  }
                  getContentAnchorEl={null}
                  id={undefined}
                  marginThreshold={0}
                  onBackdropClick={[Function]}
                  open={false}
                >
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 1
                  </li>
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 2
                  </li>
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={true}
                  >
                    Option 3
                  </li>
                  <li
                    className={undefined}
                    onClick={[Function]}
                    selected={false}
                  >
                    Option 4
                  </li>
                </WithStyles(Menu)>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Option 3",
                    <div
                      className={undefined}
                    >
                      <t
                        className="SVG-container"
                        raw="test-file-stub"
                      />
                    </div>,
                  ],
                  "className": undefined,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Option 3",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <t
                        className="SVG-container"
                        raw="test-file-stub"
                      />,
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "SVG-container",
                        "raw": "test-file-stub",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "MenuListProps": Object {
                    "className": undefined,
                    "classes": Object {
                      "root": undefined,
                    },
                    "disablePadding": true,
                  },
                  "PaperProps": Object {
                    "style": Object {
                      "background": "transparent",
                    },
                  },
                  "anchorEl": null,
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "children": Array [
                    <li
                      className={undefined}
                      onClick={[Function]}
                      selected={false}
                    >
                      Option 1
                    </li>,
                    <li
                      className={undefined}
                      onClick={[Function]}
                      selected={false}
                    >
                      Option 2
                    </li>,
                    <li
                      className={undefined}
                      onClick={[Function]}
                      selected={true}
                    >
                      Option 3
                    </li>,
                    <li
                      className={undefined}
                      onClick={[Function]}
                      selected={false}
                    >
                      Option 4
                    </li>,
                  ],
                  "getContentAnchorEl": null,
                  "id": undefined,
                  "marginThreshold": 0,
                  "onBackdropClick": [Function],
                  "open": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Option 1",
                      "className": undefined,
                      "onClick": [Function],
                      "selected": false,
                    },
                    "ref": null,
                    "rendered": "Option 1",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Option 2",
                      "className": undefined,
                      "onClick": [Function],
                      "selected": false,
                    },
                    "ref": null,
                    "rendered": "Option 2",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Option 3",
                      "className": undefined,
                      "onClick": [Function],
                      "selected": true,
                    },
                    "ref": null,
                    "rendered": "Option 3",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "3",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Option 4",
                      "className": undefined,
                      "onClick": [Function],
                      "selected": false,
                    },
                    "ref": null,
                    "rendered": "Option 4",
                    "type": "li",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
